modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -1259,6 +1259,30 @@ private static OAuthFlow cloneOAuthFlow(OAuthFlow originFlow, List<String> opera
                     }
                 }
             }
+            // This is separate than the oauth2 case since bearerAuths aren't declared in the scheme definition. They
+            // only exist at the api definition level
+            if (security != null && SecurityScheme.Type.HTTP.toString().equals(security.type)) {
+                for (SecurityRequirement requirement : securities) {
+                    List<String> opScopes = requirement.get(security.name);
+                    if (opScopes != null) {
+                        CodegenSecurity opSecurity = security.filterByScopeNames(Collections.emptyList());
+                        opSecurity.bearerFormat = security.bearerFormat;
+                        opSecurity.scheme = security.scheme;
+                        Iterator<String> it = opScopes.iterator();
+                        opSecurity.scopes = new ArrayList<>();
+                        while(it.hasNext()){
+                            Map<String, Object> scope = new HashMap<>();
+                            scope.put("scope", it.next());
+                            scope.put("hasMore", it.hasNext()? "true": null);
+                            opSecurity.scopes.add(scope);
+                        }
+                        opSecurity.hasMore = security.hasMore;
+                        result.add(opSecurity);
+                        filtered = true;
+                        break;
+                    }
+                }
+            }
```

Comment:
cc @OpenAPITools/generator-core-team for review

Reply:
I am happy to move this to a separate changeset as well so that this pr is focused on preauthorize being a capability on eligible authMethods and we can consider bearerAuth being another scheme later